src/compiler/crystal/exception.cr

```diff
@@ -254,7 +254,8 @@ module Crystal
         io << Crystal.with_line_numbers(source, line_number, @color)
       else
         from_index = [0, line_number - MACRO_LINES_TO_SHOW].max
-        source_slice = source.lines[from_index...line_number]
+        to_index = [source.lines.size, line_number].max
```

Comment:
```suggestion
        to_index = {source.lines.size, line_number}.max
```

Reply:
Why ? Has it performance benefits ?

In any case, if I do this change, would be good modify the line above too, isn't it ?

Reply:
> Why ? Has it performance benefits ?

Yes, `Tuple` is stack-allocated, unlike the heap-allocated `Array`.

Reply:
Another alternative is `Math.max` which behaves slightly differently for floating-point values (not relevant here).

Reply:
I'd suggest to clamp both values to `0..source.lines.size` to prevent any kind of issue with weird values.

Reply:
Shouldn't it be a `min`? `to_index` is the upper bound of the slice range.

Reply:
@lbguilherme Oopsie.

I'm updating with @straight-shoota suggestion.

Reply:
Just updated the code. I also fixed `from_index` value definition so it keeps consistent with `to_index`.